expression/bench_test.go

```diff
@@ -195,18 +195,57 @@ func BenchmarkScalarFunctionClone(b *testing.B) {
 	b.ReportAllocs()
 }
 
+// gener is used to generate data for test.
+type gener interface {
+	gen() interface{}
+}
+
+type rangeInt64Gener struct {
+	begin int
+	end   int
+}
+
+func (rig *rangeInt64Gener) gen() interface{} {
+	return int64(rand.Intn(rig.end-rig.begin) + rig.begin)
+}
+
+type randLenStrGener struct {
+	lenBegin int
+	lenEnd   int
+}
+
+func (g *randLenStrGener) gen() interface{} {
+	n := rand.Intn(g.lenEnd-g.lenBegin) + g.lenBegin
+	buf := make([]byte, n)
+	for i := range buf {
+		x := rand.Intn(62)
+		if x < 10 {
+			buf[i] = byte('0' + x)
+		} else if x-10 < 26 {
+			buf[i] = byte('a' + x - 10)
+		} else {
+			buf[i] = byte('A' + x - 10 - 26)
+		}
+	}
+	return string(buf)
+}
+
 type vecExprBenchCase struct {
 	retEvalType   types.EvalType
 	childrenTypes []types.EvalType
+	geners        []gener // used to generate data for children
```

Comment:
We need more comment for this, since we may only define generator for part of the children.

Reply:
more comments are added.